core/src/main/java/io/questdb/cairo/wal/seq/SeqTxnTracker.java

```diff
@@ -25,23 +25,39 @@
 package io.questdb.cairo.wal.seq;
 
 import io.questdb.cairo.ErrorTag;
+import io.questdb.cairo.wal.O3JobParallelismRegulator;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.Rnd;
 import io.questdb.std.Unsafe;
+import io.questdb.std.datetime.microtime.MicrosecondClockImpl;
 import org.jetbrains.annotations.TestOnly;
 
-public class SeqTxnTracker {
+public class SeqTxnTracker implements O3JobParallelismRegulator {
+    private static final Log LOG = LogFactory.getLog(SeqTxnTracker.class);
     private static final long SEQ_TXN_OFFSET = Unsafe.getFieldOffset(SeqTxnTracker.class, "seqTxn");
     private static final long SUSPENDED_STATE_OFFSET = Unsafe.getFieldOffset(SeqTxnTracker.class, "suspendedState");
     private static final long WRITER_TXN_OFFSET = Unsafe.getFieldOffset(SeqTxnTracker.class, "writerTxn");
+    private final Rnd rnd;
     private volatile String errorMessage = "";
     private volatile ErrorTag errorTag = ErrorTag.NONE;
+    private int maxRecordedInflightPartitions = 1;
+    // positive int: holds max parallelism
+    // negative int: holds backoff counter
+    private int regulationValue = Integer.MAX_VALUE;
```

Comment:
This is now class SeqTxnTracker. Better the filed name to indicate that it's OOM handling related
```suggestion
    private int memoryStrategyRegulationValue = Integer.MAX_VALUE;
```

Reply:
Would be clearer if the OOM handling logic lives as a member inside SeqTxnTracker but is coded as an independent class.

Reply:
We are worried about the memory footprint of `SeqTxnTracker` because they live forever, even for deleted tables. That was the reasoning behind inlining the state, and reusing `regulationValue` for two purposes.

Reply:
Rnd is still a non-primitive field